### PR TITLE
Feature/deploy django ecs

### PR DESCRIPTION
**Pull request by gecBurton**
### Key changes

* .tf for django app has been added
* `prefix` has been hardcoded

### other changes

* `tf_build_args` has been removed from `tf_apply` etc and replaced with dedicated variables
   * `core_api_image_tag`
   * `worker_image_tag`
   * `django_app_image_tag`
* various django specific environment variables that arent actually used have ben removed:
  * `AWS_STORAGE_BUCKET_NAME`
  * `AWS_S3_REGION_NAME`
  * `SENTRY_DSN`
  * `SENTRY_ENVIRONMENT`
* Beanstalk specific methods have been removed from django

## Guidance to review

Please note https://github.com/i-dot-ai/redbox-copilot/pull/292/files#r1586043620 that we have two urls atm for django-app and core-api, this is very likely to change pending https://github.com/orgs/i-dot-ai/projects/10?pane=issue&itemId=61546748

## Things to check

- [x] I have added any new ENV vars in all deployed environments
- [x] I have tested any code added or changed
- [x] I have run integration tests https://github.com/i-dot-ai/redbox-copilot/actions/runs/8921687353

### CODE REVIEW

File: django_app/redbox_app/settings.py [hunk 1]
```diff
@@ -190,37 +190,33 @@
 SESSION_COOKIE_SAMESITE = "Strict"
 SESSION_ENGINE = "django.contrib.sessions.backends.db"
 
+BUCKET_NAME = env.str("BUCKET_NAME")
```
Comment (gecBurton) [inline django_app/redbox_app/settings.py:193]: this is needed in all environments

File: infrastructure/aws/ecs.tf [hunk 2]
```diff
@@ -1,8 +1,8 @@
 locals {
   record_prefix = terraform.workspace == "prod" ? var.project_name : "${var.project_name}-${terraform.workspace}"
-  host          = "${local.record_prefix}.${var.domain_name}"
+  django_host          = "${local.record_prefix}.${var.domain_name}"
+  api_host          = "${local.record_prefix}-api.${var.domain_name}"
```
Comment (gecBurton) [inline infrastructure/aws/ecs.tf:4]: all this is very uncertain and subject to change pending a spike into how we manage https requests between apps

File: django_app/redbox_app/hosting_environment.py [hunk 3]
```diff
@@ -1,36 +1,9 @@
...
-import subprocess
-
 import environ
 
 env = environ.Env()
 
 
 class HostingEnvironment:
-    @staticmethod
-    def is_beanstalk() -> bool:
```
Comment (gecBurton) [inline django_app/redbox_app/hosting_environment.py:12]: not relevant anymore

File: Makefile [hunk 4]
```diff
@@ -161,12 +161,12 @@ tf_init: ## Initialise terraform
 .PHONY: tf_plan
 tf_plan: ## Plan terraform
 	make tf_set_workspace && \
-	terraform -chdir=./infrastructure/aws plan -var-file=$(CONFIG_DIR)/${env}-input-params.tfvars ${tf_build_args}
+	terraform -chdir=./infrastructure/aws plan -var-file=$(CONFIG_DIR)/${env}-input-params.tfvars
```
Comment (252afh) [inline Makefile:164]: We should remove the declaration of `tf_build_args` if we're not using it

File: django_app/redbox_app/settings.py [hunk 5]
```diff
@@ -5,15 +5,13 @@
 from pathlib import Path
 
 import environ
-import sentry_sdk
-from sentry_sdk.integrations.django import DjangoIntegration
+# import sentry_sdk
```
Comment (252afh) [inline django_app/redbox_app/settings.py:8]: Remove commented out bits here and the sentry bit at the bottom (if we want to keep it for future-use then move the sentry stuff outside the local setup)

**Review by 252afh — APPROVED** [review]

**Review by lmwilkigov — APPROVED** [review]

File: django_app/redbox_app/settings.py (hunk 5 above)
Comment (gecBurton) [reply django_app/redbox_app/settings.py]: [59c7228](https://github.com/i-dot-ai/redbox-copilot/pull/292/commits/59c72286dbad47ebfc927e09b39c7ca7310d1805)

File: Makefile (hunk 4 above)
Comment (gecBurton) [reply Makefile:163]: [c31cac8](https://github.com/i-dot-ai/redbox-copilot/pull/292/commits/c31cac8682bcf0dd6e137e6a8bb48bdda0b7bdc4)

**Pull request merged**